CUDADataFormats/Common/interface/SoAmacros.h

```diff
@@ -0,0 +1,297 @@
+              <<  " has size " << sizeof(TYPE) * nElements << " and padding "                                                       \
+              << ((nElements * sizeof(TYPE) / byteAlignment) + 1) * byteAlignment - (sizeof(TYPE) * nElements) <<  std::endl;       \
+  ,                                                                                                                                 \
+    std::cout << "  " BOOST_PP_STRINGIZE(NAME) "_ at offset " << offset                                                             \
+              <<  " has size " << sizeof(TYPE) << " and padding "                                                                   \
+              << (sizeof(TYPE) / byteAlignment + 1) * byteAlignment - sizeof(TYPE) <<  std::endl;                                   \
+  )
+
+#define _DECLARE_SOA_DUMP_INFO(R, DATA, TYPE_NAME)                                                                                  \
+  BOOST_PP_EXPAND(_DECLARE_SOA_DUMP_INFO_IMPL TYPE_NAME)
+
+#define _ASSIGN_SOA_COLUMN_OR_SCALAR_IMPL(IS_COLUMN, TYPE, NAME)                                                                    \
+  BOOST_PP_CAT(NAME, _) = reinterpret_cast<TYPE *>(curMem);                                                                         \
+    BOOST_PP_IIF(IS_COLUMN,                                                                                                         \
+    curMem += ((nElements_ * sizeof(TYPE) / byteAlignment_) + 1) * byteAlignment_;                                                  \
+  ,                                                                                                                                 \
+    curMem += ((sizeof(TYPE) / byteAlignment_) + 1) * byteAlignment_;                                                               \
+  )
+
+#define _ASSIGN_SOA_COLUMN_OR_SCALAR(R, DATA, TYPE_NAME)                                                                            \
+  _ASSIGN_SOA_COLUMN_OR_SCALAR_IMPL TYPE_NAME
+
+#define _ACCUMULATE_SOA_ELEMENT_IMPL(IS_COLUMN, TYPE, NAME)                                                                         \
+  BOOST_PP_IIF(IS_COLUMN,                                                                                                           \
+    ret += ((nElements * sizeof(TYPE) / byteAlignment) + 1) * byteAlignment;                                                        \
+  ,                                                                                                                                 \
+    ret += ((sizeof(TYPE) / byteAlignment) + 1) * byteAlignment;                                                                    \
+  )
+
+#define _ACCUMULATE_SOA_ELEMENT(R, DATA, TYPE_NAME)                                                                                 \
+  _ACCUMULATE_SOA_ELEMENT_IMPL TYPE_NAME
```

Comment:
This code takes care of aligning the blocks bytewise. Looking at it, a -1 is missing. I will fix that.